File: UIKit and Animations/WWDC 2014/A Look Inside Presentation Controllers (With a case study on alerts and search)/Sample_Codes/LookInside Presentation Controllers, Adaptivity, and Custom Animator Objects/LookInsidePresentationControllersAdaptivityandCustomAnimatorObjects/LookInside/AAPLPhotoCollectionViewCell.m
/*
 Copyright (C) 2014 Apple Inc. All Rights Reserved.
 See LICENSE.txt for this sample’s licensing information
 
 Abstract:
 
  AAPLPhotoCollectionViewCell implementation.
  
 */

#import "AAPLPhotoCollectionViewCell.h"

@implementation AAPLPhotoCollectionViewCell

- (instancetype)initWithFrame:(CGRect)frameRect
{
    self = [super initWithFrame:frameRect];
    if (self)
    {
        imageView = [[UIImageView alloc] init];
        [imageView setContentMode:UIViewContentModeScaleAspectFill];
        
        [[self contentView] addSubview:imageView];
        [[self contentView] setClipsToBounds:YES];
    }
    return self;
}

- (void)layoutSubviews
{
    [super layoutSubviews];
    [imageView setFrame:[[self contentView] bounds]];
}

- (void)setImage:(UIImage *)image
{
    [imageView setImage:image];
}

- (UIImage *)image
{
    return [imageView image];
}

@end

File: UIKit and Animations/WWDC 2014/A Look Inside Presentation Controllers (With a case study on alerts and search)/Sample_Codes/LookInside Presentation Controllers, Adaptivity, and Custom Animator Objects/LookInside_Swift/LookInside_Swift/AAPLPhotoCollectionViewCell.swift
//
//  AAPLPhotoCollectionViewCell.swift
//  LookInside_Swift
//
//  Created by Seyed Samad Gholamzadeh on 7/18/18.
//  Copyright © 2018 Seyed Samad Gholamzadeh. All rights reserved.
//

import UIKit

class AAPLPhotoCollectionViewCell: UICollectionViewCell {
	
	var imageView: UIImageView!
	var image: UIImage? {
		
		get {
			return self.imageView.image
		}
		
		set {
			self.imageView.image = newValue
		}
	}
	
	override init(frame: CGRect) {
		super.init(frame: frame)
		
		self.imageView = UIImageView()
		self.imageView.contentMode = .scaleAspectFill
		self.contentView.addSubview(self.imageView)
		self.contentView.clipsToBounds = true
	}
	
	required init?(coder aDecoder: NSCoder) {
		fatalError("init(coder:) has not been implemented")
	}
	
	override func layoutSubviews() {
		super.layoutSubviews()
		
		self.imageView.frame = self.contentView.bounds
	}
}
